### ISSUE TITLE
The disk guard blocks the corpus-scale runs that streaming and chunking exist to enable

### ISSUE THREAD

**Issue by thodel** [thodel/serving-atr-inference#85]
Filed from the post-mortem of `20260808T183111Z-kraken-medieval-full-v1`, which
spent 23 h in `prepare` and then died in `compile`:

```
prepare:  461,586 pages, 15,406,203 lines   (23 h, ~526 GB)
compile:  StageFailed — ketos exited -9     (after 1 h 51 m)
```

Exit -9 is SIGKILL. The leading hypothesis is the kernel OOM killer — a single
`ketos compile` over a 461 K-page manifest with `--workers 8` — but it is **not
confirmed**: `journalctl -k` implies `-b`, so a reboot since would hide it, and the
system journal needs privileges. The 111-minute climb before death fits a memory
leak, and the alternative (a service restart reaching the detached child) is ruled
out: the restarts were on 08-13, four days later.

**The OOM itself is probably already fixed.** #39 landed chunked
materialize → compile → discard on 08-10, and `_compile_chunked`
(`engines/kraken_train_svc/runner.py:104`) compiles ≤ `chunk_pages` at a time, so no
single `ketos` invocation sees the whole manifest. What this issue is about is that
**you cannot get to that path.**

## The defect

Three settings that should cooperate do not know about each other:

| setting | default | what it does |
|---|---|---|
| `cache_datasets` | **`False`** | stream parquet instead of downloading it |
| `chunk_pages` | **`0`** (off) | bound peak page-disk to one chunk |
| `min_free_disk_gb` | `50` | refuse a selection larger than this |

The size guard at `src/atr_serving/training/hf_source.py:528-552` sizes the whole
parquet selection and refuses if it exceeds `min_free_disk_gb` — **regardless of
`cache_datasets`**. But with streaming on, which is the default, those shards are
never all resident. The guard measures a quantity the configured pipeline never
materializes, and rejects on it.

The result is a closed loop: a corpus-scale selection is refused for a download that
will not happen, and the two settings that make it feasible are invisible. The
error names neither:

> the selection is ~1023.0 GB across 766 parquet shards, over the 50 GB the trainer
> keeps free. **Lower max_pages or free space.**

Lowering `max_pages` and freeing space are the two remedies that do not address it.
`ATR_TRAIN_CACHE_DATASETS=false` and `ATR_TRAIN_CHUNK_PAGES=5000` are the ones that
do, and neither is mentioned.

## And chunking is dormant

`chunk_pages: int = 0` (`settings.py:109`) means `_should_chunk`
(`runner_base.py:204`) returns `False` unless someone sets the env var. So a
corpus-scale job submitted today — if it got past the guard — would take the **same
unchunked path that died on 08-09**. The #39 machinery is complete, tested, and off.

That is also why `kraken-medieval-shard00-std` is being run by hand: sharding the
corpus manually is currently easier than switching on the feature that does it.

## Scope

- [ ] The size guard applies the parquet check only when `cache_datasets` is on.
- [ ] When streaming *unchunked* at scale, refuse with the remedy that works — page
      disk is still unbounded there, which is the 526 GB that accumulated over 23 h.
- [ ] The message names `ATR_TRAIN_CACHE_DATASETS` and `ATR_TRAIN_CHUNK_PAGES`.
- [ ] `--workers` scales down as the manifest grows, rather than a fixed 8.

## Deliberately not in scope — follow-up

**Chunked compile has no resume.** `_compile_chunked` writes `train_<i>.arrow` per
chunk and they persist, but nothing reads them back: a failure in chunk 40 restarts
from chunk 0 on a fresh job id, re-streaming and re-compiling everything. Chunking
converts one enormous loss into several smaller ones without making any of them
recoverable. `train_<i>.arrow` already existing is the natural resume marker.

Whether `chunk_pages` should default to non-zero is also left open: it changes the
path every job takes, and the 238-page test case does not need it.

**Comment by thodel** [comment]
Landed in 900d2b2 (on `main`). Four of the five scope items:

- [x] The parquet check applies only when `cache_datasets` is on.
- [x] Streaming *unchunked* at scale is still refused — page-disk is genuinely
      unbounded there — but with the remedy that works.
- [x] The message names `ATR_TRAIN_CACHE_DATASETS` and `ATR_TRAIN_CHUNK_PAGES`.
- [x] `compile --workers` tapers inverse-linearly above 20 K pages, never below 1,
      unchanged below it. `compile_workers(8, 461_586) == 1`.

A fifth case turned up while writing it: `chunk_pages` set **without**
`eval_projects` is inert, because `_should_chunk` needs them — a validation set
cannot come from splitting a stream that is discarded as it is read. That now
refuses with the actual reason instead of silently materializing everything.

693 tests pass, verified against the committed tree in a clean worktree rather than
the working copy.

**Still open, and the more valuable half:** chunked compile has no resume. Chunking
turns one enormous loss into several smaller ones without making any recoverable —
a failure in chunk 40 restarts from chunk 0 under a fresh job id, re-streaming and
re-compiling everything, even though every `train_<i>.arrow` is sitting on disk.
Whether `chunk_pages` should default to non-zero is also still open; it changes the
path every job takes.

**Comment by thodel** [comment]
Moved to https://github.com/thodel/training-atr-models/issues/26: since the split (16.09.2026) the training code lives in training-atr-models, so the work continues there. The body and comments were copied, with issue numbers rewritten to point here.
